code/memote_test_utils.py

```
@@ -0,0 +1,115 @@
+        return yml_files[0]
+    else:
+        logging.error("No XML or YML model file found in the directory.")
+        raise FileNotFoundError("No XML or YML model file found in the directory.")
+
+# Function to check and load TSV files if they exist
+def load_annotation_tsv_files(model_dir):
+    """Loads genes.tsv, mets.tsv, and rxns.tsv if all exist, otherwise raises an error."""
+    required_tsv_files = ["genes.tsv", "mets.tsv", "rxns.tsv"]
+    tsv_paths = {tsv: os.path.join(model_dir, tsv) for tsv in required_tsv_files}
+
+    missing_files = [tsv for tsv, path in tsv_paths.items() if not os.path.exists(path)]
+    if missing_files:
+        logging.error(f"Missing required annotation file(s): {', '.join(missing_files)}")
+        raise FileNotFoundError(f"Missing required annotation file(s): {', '.join(missing_files)}")
+
+    logging.info("Annotation TSV information loaded.")
+    return {tsv: pd.read_csv(path, sep="\t") for tsv, path in tsv_paths.items()}
+
+# Function to parse annotation field
+def parse_annotation_field(field_str):
+    """Parses a structured annotation field (e.g., NAME, MIRIAM) into a dictionary."""
+    annotation_dict = {}
+    if pd.notna(field_str):
+        for entry in field_str.split(";"):
+            key_value = entry.split("/")
+            if len(key_value) == 2:
+                key, value = key_value
+                annotation_dict.setdefault(key, []).append(value)
+    return annotation_dict
+
+# Function to append annotation information to the model
+def append_annotations(model, genes_df, mets_df, rxns_df):
+    """Adds annotation data to the model from TSV files."""
+    # Ensure all genes have SBO:0000243
+    for gene in model.genes:
+        gene.annotation.setdefault("sbo", "SBO:0000243")
+        
+    # Update gene annotations
+    for _, row in genes_df.iterrows():
+        gene_id_clean = row[genes_df.columns[0]].removeprefix("G_")  # Cleaned ID
+        if gene_id_clean in model.genes:
```

Do we want to raise an exception if the gene_id_clean is not present in the model?

Why do we prefer to run through the id's via the tsv files instead of the model-id's?

---

It would be an improvement raise an exeption or have the code hard-fail in that case. We run through the id's via the tsv files out of convenience, not preference. Because this workflow will shortly become deprecated it is unnecessary to implement any changes at this time.